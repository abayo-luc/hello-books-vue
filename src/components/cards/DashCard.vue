<template>
  <div
    class="dash-card"
    :style="{backgroundImage: 'url(\'' + require('../../assets/' + img ) + '\')'}"
  >
    <div class="container">
      <p>{{title}}</p>
    </div>
  </div>
</template>
<script>
export default {
  name: 'DashCard',
  props: {
    title: {
      type: String,
      required: true,
    },
    img: {
      type: String,
      required: true,
    },
  },
};
</script>
<style lang="scss" scoped>
.dash-card {
  border-radius: 15px;
  box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.08);
  transition: 0.3s;
  width: 23%;
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  cursor: pointer;
  &:hover {
    box-shadow: 0 10px 18px 0 rgba(0, 0, 0, 0.2);
  }
  .container {
    height: 5em;
    display: flex;
    padding: 15px;
    justify-content: flex-start;
    align-items: flex-end;
    color: #fff;
    p {
      text-transform: capitalize;
      font-weight: 600;
    }
  }
}
</style>
