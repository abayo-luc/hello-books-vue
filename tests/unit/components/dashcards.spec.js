import {
  mount
} from '@vue/test-utils';
import DashCard from '../../../src/components/DashCards.vue';


const wrapper = mount(DashCard);

describe('DashCard.vue', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
