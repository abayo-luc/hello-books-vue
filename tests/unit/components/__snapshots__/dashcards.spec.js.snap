// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashCard.vue should match snapshot 1`] = `
<div class="dash-cards">
  <div class="dash-card" style="background-image: url();">
    <div class="container">
      <p>New Books</p>
    </div>
  </div>
  <div class="dash-card" style="background-image: url();">
    <div class="container">
      <p>Top books of 2019</p>
    </div>
  </div>
  <div class="dash-card" style="background-image: url();">
    <div class="container">
      <p>Recomendations</p>
    </div>
  </div>
  <div class="dash-card" style="background-image: url();">
    <div class="container">
      <p>Your reading list</p>
    </div>
  </div>
</div>
`;
